pkg/services/pluginsintegration/clientmiddleware/forward_id_middleware.go

```diff
@@ -32,6 +36,15 @@ func (m *ForwardIDMiddleware) applyToken(ctx context.Context, pCtx backend.Plugi
 		return nil
 	}
 
+	jsonDataBytes, err := simplejson.NewJson(pCtx.DataSourceInstanceSettings.JSONData)
```

Comment:
also, you probably need to check that `pCtx.DataSourceInstanceSettings` is not nil in the condition above

Reply:
yes, there are already multiple middlewares doing the same which is why it could be a bit worrying but it's probably something that we need to tackle separately